src/main/kotlin/org/rust/lang/core/RustPsiPattern.kt

```diff
@@ -80,6 +81,12 @@ object RustPsiPattern {
     val onTestFn: PsiElementPattern.Capture<PsiElement> = onItem(psiElement<RustFunctionElement>()
         .withChild(psiElement<RustOuterAttrElement>().withText("#[test]")))
 
+    val inAnyLoop: PsiElementPattern.Capture<PsiElement> =
+        psiElement().inside(true, psiElement<RustBlockElement>().withParent(or(
```

Comment:
One possible improvement in performance and correctness is to try to stop earlier when searching for parent. That is, `break` should not be suggested inside a lambda inside a loop.